### PR TITLE
Fix Element handling from within Categories tree

### PR DESCRIPTION
**Pull request by smg6511**
### What does it do?
Adds a new tree method to properly extract Element data from the working tree Node. There are three patterns for a Node's id, but only one was being accounted for in the code. Additionally, cleans up code formatting in the affected methods.

### Why is it needed?
Within the Categories tree, these actions are currently inoperable:
- Element removal
- Plugin activation/deactivation

### How to test
1. Create a handful of Elements (at least one of each type), both within a Category and within the root (no Category)
2. Verify the ability to remove each Element (from both the Categories tree and from within the individual Element trees)
3. Verify the ability to activate/deactivate Plugins (from both the Categories tree and from within the individual Element trees)

### Related issue(s)/PR(s)
Resolves https://github.com/modxcms/revolution/issues/16487 for MODX 2.x
Port of PR https://github.com/modxcms/revolution/pull/16489

### CODE REVIEW

**Review by theboxer — DISMISSED** [review]
👍 thanks for the port to 2.x

**Review by opengeek — CHANGES_REQUESTED** [review]
Attempting to run the grunt build on this PR fails:

```Running "uglify:jsgrps" (uglify) task
JS_Parse_Error [SyntaxError]: Invalid assignment
    at JS_Parse_Error.get (eval at <anonymous> (/Users/opengeek/www/revo-2.x/_build/templates/default/node_modules/uglify-js/tools/node.js:18:1), <anonymous>:71:23)
    at formatError (internal/util/inspect.js:1149:38)
    at formatRaw (internal/util/inspect.js:919:14)
    at formatValue (internal/util/inspect.js:774:10)
    at inspect (internal/util/inspect.js:319:10)
    at formatWithOptionsInternal (internal/util/inspect.js:1979:40)
    at formatWithOptions (internal/util/inspect.js:1861:10)
    at console.value (internal/console/constructor.js:328:14)
    at console.log (internal/console/constructor.js:364:61)
    at /Users/opengeek/www/revo-2.x/_build/templates/default/node_modules/grunt-contrib-uglify/tasks/uglify.js:144:17 {
  filename: 'widgets/element/modx.tree.element.js',
  line: 213,
  col: 29,
  pos: 7614
}
```

File: manager/assets/modext/widgets/element/modx.tree.element.js [hunk 1]
```diff
@@ -179,62 +179,123 @@ Ext.extend(MODx.tree.Element,MODx.tree.Tree,{
...
+        const identifiers = activeNode.id.substr(startIndex).split('_');
+
+        /*
+            Expected array items:
+            - When working in the Categories tree: [element type, node type ('element'), element id, element's category id]
+            - When working in any of the five Element trees: [element type, node type ('category'), element's category id]
+            - When creating and Element in the root of it's type's tree: [element type]
+        */
+
+        [extractedData.type] = identifiers;
```
Comment (Mark-H) [inline manager/assets/modext/widgets/element/modx.tree.element.js:213]: ^

**Comment by smg6511** [comment]
Guys, I see what's happening here and on another one of the 2.x builds that's choking (#16467): We'd never updated the template build config like we did for 3.x. Unless there's opposition to doing so, let me go ahead and back port the 3.x setup to 2.x ... then we won't have these issues and won't have to handle code differently between the two branches.

File: manager/assets/modext/widgets/element/modx.tree.element.js (hunk 1 above)
Comment (opengeek) [reply manager/assets/modext/widgets/element/modx.tree.element.js:213]: Can a work around be put in place for this so I can get a release out? We are way overdue on a release.

**Comment by opengeek** [comment]
> Guys, I see what's happening here and on another one of the 2.x builds that's choking (#16467): We'd never updated the template build config like we did for 3.x. Unless there's opposition to doing so, let me go ahead and back port the 3.x setup to 2.x ... then we won't have these issues and won't have to handle code differently between the two branches.

There's no opposition. Is this something you can get done soon? I'd like to get a release out ASAP.

**Pull request merged**